analyzers/ImpactParameter.py

```diff
@@ -1,48 +1,233 @@
 from heppy.framework.analyzer import Analyzer
-from ROOT import TVector3
-#from heppy.framework.event import Event
+from ROOT import TFile, TH1F
+from ROOT import TVector3, TLorentzVector
 from heppy.papas.path import Helix
-#import collections
+import math
+from heppy.utils.deltar import deltaR
 
 class ImpactParameter(Analyzer):
-
+    '''This analyzer puts an impact parameter for every charged particle
+    as an attribute of its path : particle.path.IP
+    The significance is calculated, the calculus are a first order approximation,
+    thus this may not be correct for large impact parameters (more than 3 mm)
+    It also computes W, the likelihood value for the jet to contain beauty,
+    using a likelihood ratio method for each particle in the jet.
+    Computes TCHE and TCHP for the jet for b-tagging'''
         
     def process(self, event):
         assumed_vertex = TVector3(0, 0, 0)
         jets = getattr(event, self.cfg_ana.jets)
+        enable_matter_scattering = self.cfg_ana.enable_matter_scattering
         for jet in jets:
-            #for ptc in jet.constituents[211]:
-            #    IPsn = 1
-            #    N_charge = 0
-                #for i in range(len(const)):
-                #    if const[i][1][0]._charge <> 0:
-                #        if len const[i][1] > 0:
-                #            for l in range(len(const[i][1]))
-                #                partic = const[i][1][l]
-                #                h = partic.path
-                #                Helix.compute_IP(h, assumed_vertex)
-                #                setattr(partic, self.IP, h.IP)
-                #                setattr(partic, self.tIP, h.closest.t)
-                #                setattr(partic, self.xIP, h.closest.x)
-                #                setattr(partic, self.yIP, h.closest.y)
-                #                setattr(partic, self.zIP, h.closest.z)
-                #                IPsn *= h.IP
-                #                N_charge += 1
-                #        else :
-                #            pass
-                #    else :
-                #        pass
-                #if N_charge <> 0 :
-                #    IPs = IPSn**(1.0/N_charge)
-                #else :
-                #    pass
+            IP_b_LL = 0
+            IPs_b_LL = 0
+            ptcs_pt = []
```

Comment:
all these arrays are a sign that an object or a structure should hold all this information.